ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlockProcessorUtilTest.java

```diff
@@ -42,18 +46,31 @@
   private final DataStructureUtil dataStructureUtil = new DataStructureUtil();
 
   @Test
-  @Disabled
   void processDepositAddsNewValidatorWhenPubkeyIsNotFoundInRegistry()
       throws BlockProcessingException {
-    // Data Setup
-    SSZList<DepositWithIndex> deposits = dataStructureUtil.newDeposits(1);
-    Deposit deposit = deposits.get(0);
-    DepositData depositInput = deposit.getData();
+    // Create a deposit
+    DepositData depositInput = dataStructureUtil.randomDepositData();
     BLSPublicKey pubkey = depositInput.getPubkey();
     Bytes32 withdrawalCredentials = depositInput.getWithdrawal_credentials();
-    UnsignedLong amount = deposit.getData().getAmount();
-
-    BeaconState beaconState = createBeaconState();
+    UnsignedLong amount = depositInput.getAmount();
+
+    // Add the deposit to a Merkle tree so that we can get the root to put into the state
+    MerkleTree depositMerkleTree = new OptimizedMerkleTree(Constants.DEPOSIT_CONTRACT_TREE_DEPTH);
+    depositMerkleTree.add(depositInput.hash_tree_root());
+
+    // Create the state and insert the Merkle root of the deposit data
+    BeaconState beaconState =
+        createBeaconState()
+            .updated(
+                state ->
+                    state.setEth1_data(
+                        new Eth1Data(
+                            depositMerkleTree.getRoot(), UnsignedLong.valueOf(1), Bytes32.ZERO)));
+
+    SSZMutableList<DepositWithIndex> deposits =
+        SSZList.createMutable(DepositWithIndex.class, Constants.MAX_DEPOSITS);
+    deposits.add(
+        new DepositWithIndex(depositMerkleTree.getProof(0), depositInput, UnsignedLong.valueOf(0)));
```

Comment:
(optional) Might be worth trying to dedupe some of this logic with a helper method that sets up and processes the state & deposits and returns the resulting state.

Reply:
Heh, busted. Yes, I was too lazy to do this. Now done.